### PR TITLE
Redux auth

### LINKED ISSUE #9: Implement redux to ease state handling
Currently, auth state is being handled by state lifting. Redux is a viable option for consistent state management. Implement that in existing functionality.

### PR DESCRIPTION
**Pull request by ShubhankarKG**
Fixes #9 
Basic redux folder structure.

### CODE REVIEW

**Comment by ShubhankarKG** [comment]
Rebased master with current branch head and resolved merge conflicts

File: new_client/src/actions/authActions.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  REGISTER_FAIL
+} from "./types";
+import axios from "axios";
+import { returnErrors } from "./errorActions";
+
+export const loadUser = () => (dispatch, getState) => {
+  dispatch({ type: USER_LOADING });
+
+  axios
+    .get("http://localhost:5000/api/user", tokenConfig(getState))
```
Comment (AliabbasMerchant) [inline new_client/src/actions/authActions.js:17]: Use the environment variable here. Also, The env variable defines the port as something else, and you have defined it as 5000. Hence, use the env. variable itself, for the base url of the server/API

File: new_client/src/actions/authActions.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+export const register = ({ username, email, password, graduationYear }) => (dispatch) => {
+  const config = {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+  const body = JSON.stringify({ username, email, password, graduationYear });
+  console.log(body);
+  console.log(process.env.REACT_APP_API_REGISTER)
```
Comment (AliabbasMerchant) [inline new_client/src/actions/authActions.js:39]: It is definitely okay to log your debugging variables when testing/trying out stuff. But when making a PR, please make sure to remove such logs, unless they are important.

File: new_client/src/actions/authActions.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    .post(process.env.REACT_APP_API_LOGIN, body, config)
+    .then(res =>
+      dispatch({
+        type: LOGIN_SUCCESS,
+        payload: res.data
+      })
+    )
+    .catch(err => {
+      dispatch(
+        returnErrors(err.response.data, err.response.status, "LOGIN_FAIL")
```
Comment (AliabbasMerchant) [inline new_client/src/actions/authActions.js:78]: Here, "LOGIN_FAIL" is the id, right?
Also, it is a special-valued string used across files. So, in order to reduce errors, declare a const variable somewhere, and use that across the files. (The `LOGIN_FAIL` variable is already declared in the types file. You can directly use that)

File: new_client/src/actions/types.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export const REGISTER_USER = "REGISTER_USER";
```
Comment (AliabbasMerchant) [inline new_client/src/actions/types.js:1]: Not exactly required, but I think using Redux Toolkit is definitely better and easier

File: new_client/src/components/Header.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ export default function Header() {
         <Link to="/"><i className="fa fa-fw fa-info"></i>EVENTS</Link>
         <Link to="/glimpse"><i className="fa fa-fw fa-info"></i>GLIMPSES</Link>
         <Link to="/blogs"><i className="fa fa-fw fa-info"></i>BLOGS</Link>
-        <Link to="/auth"><i className="fa fa-fw fa-info"></i> SIGN UP</Link>
+        <Link to="/signup"><i className="fa fa-fw fa-info"></i> SIGN UP</Link>
```
Comment (AliabbasMerchant) [inline new_client/src/components/Header.js:16]: Good. `/signup` is a more intuitive route name, than `/auth`

File: new_client/src/components/auth/Signin.jsx [hunk 6]
```diff
@@ -1,12 +1,20 @@
-import React from 'react';
-import { Container, Paper, Grid, TextField, Button } from '@material-ui/core';
+import React, {useState, useEffect} from 'react';
+import { Container, Paper, Grid, TextField, Button, Typography, Link } from '@material-ui/core';
```
Comment (AliabbasMerchant) [inline new_client/src/components/auth/Signin.jsx:2]: Again, personal opinion, so not so sure, but shouldn't we stick to 1 UI library? Somewhere we are using MaterialUI, and someplaces we are using Bootstrap. @PrabhuPratik27 What do you feel?

File: new_client/src/reducers/errorReducer.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { GET_ERRORS, CLEAR_ERRORS } from "../actions/types";
+
+const initialState = {
+  msg: {},
+  status: null,
+  id: null,
+};
+
+export default function(state=initialState, action) {
```
Comment (AliabbasMerchant) [inline new_client/src/reducers/errorReducer.js:9]: Try to always name your exported functions. Its helpful during debugging

File: new_client/src/store/configureStore.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+  if (module.hot) {
+    module.hot.accept('../reducers', () => {
+      const nextRootReducer = require('../reducers').default;
+      store.replaceReducer(nextRootReducer);
+    })
+  }
+
+  return store;
+}
```
Comment (AliabbasMerchant) [inline new_client/src/store/configureStore.js:24]: Again, Redux Toolkit does all this automatically

File: new_client/src/actions/types.js (hunk 4 above)
Comment (ShubhankarKG) [reply new_client/src/actions/types.js:1]: It was decided (within the SYs) to use Redux only, hence used.

Comment (AliabbasMerchant) [reply new_client/src/actions/types.js:1]: Ok

File: new_client/src/App.js [hunk 9]
```diff
@@ -1,27 +1,23 @@
...
+import Home from './components/Home/home';
+import Footer from './components/Footer'
+import About from './components/pages/About';
+import Glimpse from './components/glimpses/Glimpse';
+
+import Blogs from './components/blogs/Blog';
+import AddBlog from './components/blogs/AddBlog';
+
 import Signin from './components/auth/Signin.jsx';
 import Signup from './components/auth/Signup.jsx';
```
Comment (AliabbasMerchant) [inline new_client/src/App.js]: What about these 2?

File: new_client/src/App.js [hunk 10]
```diff
@@ -30,15 +26,16 @@ function App() {
...
-          <Route path="/signin" component={Signin} />
-          <Route exact path="/addblog" component={Addblog} />
-          <Route path="/signup" component={Signup} />
-
-          <Route path="/glimpse" component={Glimpse} />
-          {/* <Footer /> */}
+          <Switch>
+            <Route exact path="/" component={Home} />
+            <Route path="/about" component={About} />
+            <Route exact path="/blogs" render={() => <Blogs />} />
```
Comment (AliabbasMerchant) [inline new_client/src/App.js:32]: We can now directly put Blogs in the component prop

**Pull request merged**